assert/assertions.go

```diff
@@ -1565,6 +1565,65 @@ func JSONEq(t TestingT, expected string, actual string, msgAndArgs ...interface{
 	return Equal(t, expectedJSONAsInterface, actualJSONAsInterface, msgAndArgs...)
 }
 
+// JSONPartialEq asserts that the expected JSON construct exists within
+// the actual. Note that while the order of the properties does not
+// matter, the order of elements within an array does matter. This can
+// be circumvented by using an empty map to properly offset the expected.
+//
+//    assert.JSONPartialEq(t, `{"foo": {"foo": "bar"}}`, `{"hello": "world", "foo": {"foo": "bar", "hello": "world"}}`)
+//    assert.JSONPartialEq(t, `{"foo": [{}, {"hello": "world"}]}`, `{"hello": "world", "foo": [{"foo": "bar"}, {"hello": "world"}]}`)
+func JSONPartialEq(t TestingT, expected string, actual string, msgAndArgs ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+
+	var expectedJSONAsInterface, actualJSONAsInterface interface{}
+
+	if err := json.Unmarshal([]byte(expected), &expectedJSONAsInterface); err != nil {
+		return Fail(t, fmt.Sprintf("Expected value ('%s') is not valid json.\nJSON parsing error: '%s'", expected, err.Error()), msgAndArgs...)
+	}
+
+	if err := json.Unmarshal([]byte(actual), &actualJSONAsInterface); err != nil {
+		return Fail(t, fmt.Sprintf("Input ('%s') needs to be valid json.\nJSON parsing error: '%s'", actual, err.Error()), msgAndArgs...)
+	}
+
+	prune(&actualJSONAsInterface, &expectedJSONAsInterface)
+
+	return Equal(t, expectedJSONAsInterface, actualJSONAsInterface, msgAndArgs...)
+}
+
+// Remove all map key/values that exist in `a` but not in `b`
+// (unordered), and remove all array elements that exist in `a`
+// but not in `b` (ordered).
+func prune(a, b *interface{}) {
+	switch aTyped := (*a).(type) {
+	case map[string]interface{}:
+		if bMap, ok := (*b).(map[string]interface{}); ok {
+			for k, aValue := range aTyped {
+				if bValue, ok := bMap[k]; ok {
+					prune(&aValue, &bValue)
+					aTyped[k] = aValue
+				} else {
+					delete(aTyped, k)
+				}
```

Comment:
We gave it a try in our local setup. But we removed this because it was hard to see exactly what was passed since anything that wasn't specified is showing up in the assertion message, so we can't tell what we got and compared it to what was expected.

Reply:
It seems we can check them partially without pruning them. Laravel has implemented a similar assert function.
See the following link:
https://github.com/laravel/framework/blob/b9203fca96960ef9cd8860cb4ec99d1279353a8d/src/Illuminate/Testing/TestResponse.php#L772

Reply:
The reason that I chose to prune was so that I could reuse an existing method to do the actual JSON compare. If you remove the pruning (specifically the key/value deletion) it may not correctly compare the two JSON objects based on the unit tests that are included.

The alternative is to write a partial JSON comparison from the ground up without relying on `Equal()`.

Reply:
@alan2wong I see, maybe that is exactly what we needed or trying to end up doing since the intention was to ignore Timestamps and IDs from the comparison since they aren't important.